decode/layout.ts

```diff
@@ -156,7 +158,10 @@ function getHash(id: number, tag: string, path: string, attributes: Attributes):
             if ("id" in attributes) { s = `${tag}#${attributes["id"]}`; }
             if ("class" in attributes) { s += `.${attributes["class"].trim().split(" ").join(".")}`; }
             if (ID_ATTRIBUTE in attributes) { s = `*${attributes[ID_ATTRIBUTE]}`; }
-            if (s) { hashes.push({ id, hash: generateHash(s), selector: s }); }
+            if (s) {
+                hashLookup[id] = hashes.length;
```

Comment:
Why do you need two tables (hashlookup and hashes)? Could you not combine into a single table?